File: quarkus-jdbi/src/main/java/com/github/frederikpietzko/dto/CreateVisitDto.java
package com.github.frederikpietzko.dto;

import com.github.frederikpietzko.model.Visit;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;

import java.time.OffsetDateTime;

public record CreateVisitDto(
  @NotEmpty String description,
  @NotNull CreatePetDto pet,
  @NotNull OffsetDateTime dateTime
) {

  public Visit toVisitEntity() {
    return new Visit(null, description, pet.toEntity(), dateTime);
  }
}

File: http4k/src/main/kotlin/com/github/frederikpietzko/dto/CreateVisitDto.kt
package com.github.frederikpietzko.dto

import kotlinx.serialization.Serializable
import java.time.OffsetDateTime

@Serializable
data class CreateVisitDto(
  val description: String,
  val pet: CreatePetDto,
  @Serializable(with = OffsetDateTimeSerializer::class) val dateTime: OffsetDateTime,
) {
  fun toModel() = com.github.frederikpietzko.model.Visit(
    id = null,
    description = description,
    pet = pet.toModel(),
    dateTime = dateTime,
  )
}

